docs/book/src/developer/providers/migrations/v1.10-to-v1.11.md

```diff
@@ -146,38 +360,127 @@ support for the v1beta1 version of the Cluster API contract will be removed tent
 After compatibility support for the v1beta1 version of the Cluster API contract will be removed, providers
 which are still implementing the v1beta1 contract will stop to work (they will work only with older versions of Cluster API).
 
-See [provider contracts](../contracts/overview.md) for more details.
+Please see [Cluster API Contract changes](#cluster-api-contract-changes) and [provider contracts](../contracts/overview.md) for more details.
 
-#### How to bump to CAPI V1.11 but keep using deprecated v1beta1 conditions
+### How to implement the new v1beta2 contract
 
-A provider can continue to use deprecated v1beta1 conditions also after bumping to CAPI V1.11.
+We highly recommend providers to start planning the move to the new v1beta2 version of the Cluster API contract as soon
+as possible.
 
-However:
-- It is required to use conditions type from the v1beta2 API version in order to keep using conditions utils.
-- Conditions utils for deprecated v1beta1 conditions have been moved to `util/conditions/deprecated/v1beta1`
+Implementing the new v1beta2 contract for providers is a two step operation:
+1. Implement changes defined for a specific provider type; See [Cluster API Contract changes](#cluster-api-contract-changes) and [provider contracts](../contracts/overview.md) for more details.
+  - In most cases, v1beta2 contract introduced changes in the `initialization completed`, `conditions`, `terminal failures` rules;
+    Also `replicas` rule is changed for control plane providers.
+2. While implementing the , It is also highly recommended to check the implementation of all the other rules
+   (documentation for contract rules have been improved in recent releases, worth to take a look!).
+3. Change the CRD annotation that document which Cluster API contract is implemented by your Provider. e.g.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    cluster.x-k8s.io/v1beta2: <your API version>
+  name: <your CRD name>
+```
+
+### How to bump to CAPI V1.11 but keep using deprecated v1beta1 conditions
+
+A provider can continue to use deprecated v1beta1 conditions also after bumping to CAPI V1.11, but to do
+it is require to change following imports:
+
+```go
```

Comment:
Does it work correctly ?
It looks like that syntax highlight is broken.

Reply:
It doesn't seems bad to me
<img width="932" alt="image" src="https://github.com/user-attachments/assets/f7a8f154-bcab-4183-ada5-d5a4ffbb69cd" />